homeassistant/components/telegram_bot/bot.py

```diff
@@ -287,24 +287,19 @@ def _get_msg_ids(
             inline_message_id = msg_data["inline_message_id"]
         return message_id, inline_message_id
 
-    def _get_target_chat_ids(self, target: Any) -> list[int]:
+    def _get_target_chat_ids(self, target: int | list[int] | None) -> list[int]:
         """Validate chat_id targets or return default target (first).
 
         :param target: optional list of integers ([12234, -12345])
         :return list of chat_id targets (integers)
         """
         allowed_chat_ids: list[int] = self._get_allowed_chat_ids()
-        default_user: int = allowed_chat_ids[0]
-        if target is not None:
-            if isinstance(target, int):
-                target = [target]
-            chat_ids = [t for t in target if t in allowed_chat_ids]
-            if chat_ids:
-                return chat_ids
-            _LOGGER.warning(
-                "Disallowed targets: %s, using default: %s", target, default_user
-            )
-        return [default_user]
+
+        if target is None:
+            return [allowed_chat_ids[0]]
+
+        chat_ids = [target] if isinstance(target, int) else target
+        return [chat_id for chat_id in chat_ids if chat_id in allowed_chat_ids]
```

Comment:
Does it matter if the list is empty that we return from `_get_target_chat_ids`? We guarded for that before but don't do that now.

Reply:
Ok.

An edge case seems to be a from the beginning empty target list which will not raise HomeAssistantError on line 411 since failed_chat_ids will also be empty and falsy.

We don't require at least one item in the target list at the moment. Maybe that is something to consider for the future?

It could be confusing to the user and hard to troubleshoot as we don't give any feedback for that case which most likely is a user error.

Reply:
The target parameter is optional. Currently, and with this PR, a not set target parameter and a target parameter with empty list means the same thing. Maybe that's ok and we don't need to change anything, or we think it could be confusing for users? I just wanted to mention it.

Reply:
Correction: It doesn't mean the same thing for what chat ID will be used but neither will cause an error.

Not setting the target parameter will use the first (default) chat ID, while passing an empty list will use no chat ID at all.

Reply:
You're right and I confused the two scenarios: empty vs not set. I agree with you now: an empty target list is probably a user error.